### DISCUSSION TITLE
Lazy loading components

### DISCUSSION THREAD

**Discussion by aitorllj93** [orchestratora/orchestrator#75] [Ideas]
Hi, I'm really interested in this project. I have a few questions about the current state of the dynamic loading engine. Does this support lazy loading standalone components and/or [webpack federated modules](https://github.com/angular-architects/module-federation-plugin)?

**Comment by gund** [comment]
Hey, at the moment the core does not support any lazy loading of components and I want to add support for it at some point.

However this does not really prevent you from implementing a custom component that lazy loads real component and renders it.
For now you would have to implement rendering on your own via `*ngComponentOutlet` directive or [`ng-dynamic-component`](https://github.com/gund/ng-dynamic-component) lib if you need to pass inputs/outputs.

In the future I would like to allow to register components at runtime in [`ComponentLocatorService`](https://github.com/orchestratora/orchestrator/blob/8eb225cafa10ae71c5d5c0c00cfa0376459f90d9/libs/core/src/lib/component-locator/component-locator.service.ts#L21) instead of inject/compile time which will allow to use `<orc-render-item>` component directly to render dynamically registered components like so:
https://github.com/orchestratora/orchestrator/blob/6d0d2bfd0b071edd06f62b8ae90e83ea2513f1b0/libs/core/src/lib/orchestrator/orchestrator.component.html#L1-L6

Also RFCs and/or PRs are welcome 😊

**Comment by gund** [comment]
I've created an RFC for this here https://github.com/orchestratora/rfcs/pull/5
Please feel free to add your input/feedback if you are interested.
